aesd-char-driver/main.c

```diff
@@ -0,0 +1,286 @@
+/**
+ * @file aesdchar.c
+ * @brief Functions and data related to the AESD char driver implementation
+ *
+ * Based on the implementation of the "scull" device driver, found in
+ * Linux Device Drivers example code.
+ *
+ * @author Dan Walkes
+ * @date 2019-10-22
+ * @copyright Copyright (c) 2019
+ *
+ */
+
+#include <linux/module.h>
+#include <linux/syscalls.h>
+#include <linux/init.h>
+#include <linux/printk.h>
+#include <linux/types.h>
+#include <linux/cdev.h>
+#include <linux/fs.h> // file_operations
+#include <asm/uaccess.h>
+#include "aesdchar.h"
+int aesd_major =   0; // use dynamic major
+int aesd_minor =   0;
+
+MODULE_AUTHOR("Jordi Cros Mompart");
+MODULE_LICENSE("Dual BSD/GPL");
+
+struct aesd_dev aesd_device;
+
+int aesd_open(struct inode *inode, struct file *filp)
+{
+    //Handle open
+    PDEBUG("open\n");
+    try_module_get(THIS_MODULE);
```

Comment:
@jordicrosmompart I'm curious where you found this, it shouldn't be necessary.  See https://stackoverflow.com/questions/1741415/linux-kernel-modules-when-to-use-try-module-get-module-put/6079839#6079839